pkg/tsdb/cloudwatch/get_dimension_values_for_wildcards.go

```diff
@@ -0,0 +1,57 @@
+package cloudwatch
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/clients"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models/resources"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/services"
+	"github.com/patrickmn/go-cache"
+)
+
+// getDimensionValues gets the actual dimension values for dimensions with a wildcard
+func (e *cloudWatchExecutor) getDimensionValuesForWildcards(ctx context.Context, client models.CloudWatchMetricsAPIProvider, region string, queries []*models.CloudWatchQuery) ([]*models.CloudWatchQuery, error) {
+	metricsClient := clients.NewMetricsClient(client, e.cfg)
+	service := services.NewListMetricsService(metricsClient)
+
+	for _, query := range queries {
+		for dimensionKey, values := range query.Dimensions {
+			// if the dimension is not a wildcard, skip it
+			if len(values) != 1 || query.MatchExact || (len(values) == 1 && values[0] != "*") {
+				continue
+			}
+
+			cacheKey := fmt.Sprintf("%s-%s-%s-%s", region, query.Namespace, query.MetricName, dimensionKey)
```

Comment:
Good call on the security, I moved the cache to be by instance so that you have to have the related permissions to access the cache. I should have done that anyway to avoid causing future multi-tenancy issues.

I don't think labels should change frequently. 24 hours of expiration gets us to 30 queries a month per dimension, and we could up it a bit. I'm not sure what the best expiration would be, but 1 hour or 720 queries per dimension probably wouldn't be crazy out of the 1 million. We could make it configurable? They could resave the datasource to clear the cache if they need, though I don't know if all users would have access to that.